### ISSUE TITLE
Add a verify script that checks for json tags in versioned api types.go

### ISSUE THREAD

**Issue by pmorie** [kubernetes-incubator/service-catalog#446]
All fields in `types.go` files in versioned APIs should have a json tag.  This is a candidate to pull into repo-infra from k8s.

**Comment by jpeeler** [comment]
I looked at this and attempted to verify that hack/verify-description.sh is the script we need to import from k8s. But after removing a JSON tag in pkg/api/v1/types.go and then testing, the script didn't fail. When running "make verify", I didn't see any output specifically illustrating the diff I was expecting so maybe I'm testing incorrectly? update-codecgen.sh at least produced a diff that something changed from the previous working state, but this doesn't seem to match what this issue is requesting.

**Comment by duglin** [comment]
@jpeeler Here's the start of a `verify-types.go` file that will check to make sure "json:" appears in the tag of each struct field:

```
package main

import (
	"fmt"
	"go/ast"
	"go/parser"
	"go/token"
	"os"
	"strings"
)

func verifyTypes(fileName string) int {
	rc := 0
	fset := token.NewFileSet()

	fileAst, err := parser.ParseFile(fset, fileName, nil, parser.ParseComments)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error parsing file %q: %s\n", fileName, err)
		return 1
	}

	for _, decl := range fileAst.Decls {
		genD, ok := decl.(*ast.GenDecl)

		// Skip non-Type decls
		if !ok || genD.Tok != token.TYPE {
			continue
		}

		for _, spec := range genD.Specs {
			daType, ok := spec.(*ast.TypeSpec)
			if !ok {
				continue
			}

			// Skip non-struct types
			structType, ok := daType.Type.(*ast.StructType)
			if !ok {
				continue
			}

			if structType.Fields != nil {
				for _, field := range structType.Fields.List {
					// Note that we'll skip anonymous fields
					for _, name := range field.Names {
						// No tag, not a string or no "json:" and we error it!
						if field.Tag == nil ||
							field.Tag.Kind != token.STRING ||
							!strings.Contains(field.Tag.Value, "json:") {

							fmt.Printf("%s.%s: missing json tag\n",
								daType.Name, name)
							rc = 1
						}
					}
				}
			}
		}
	}

	return rc
}

func main() {
	rc := 0
	for _, fn := range os.Args[1:] {
		rc += verifyTypes(fn)
	}

	os.Exit(rc)
}
```

I think we'd just need to call it for each of our versioned types golang files.

**Comment by jpeeler** [comment]
Looks nice and might be small enough to just keep in service-catalog, but I'd expect this sort of validation to live in k8s for everybody to use eventually.

**Comment by duglin** [comment]
yea, we can add it to: https://github.com/kubernetes/repo-infra  once we know it works

**Comment by MHBauer** [comment]
closed by #1480
